Sources/NIOPosix/PendingDatagramWritesManager.swift

```diff
@@ -381,7 +381,10 @@ extension PendingDatagramWritesState {
 /// the availability of the functions.
 final class PendingDatagramWritesManager: PendingWritesManager {
 
+    /// storage for IOVector and the references to the buffers used when we perform gathering writes. Only present
+    /// on Linux because Darwin does not support gathering datagram writes.
     private var bufferPool: Pool<PooledBuffer>
+    private var buffer: PooledBuffer?
```

Comment:
Why do we need the local buffer reference?

Reply:
There are 2 reasons for that:
1) we will need to put that pooled buffer back to the pool when async operation will be completed
2) we should keep a memory block which is used for the async operation while the operation will not be completed

And we also have 2 options how to do it:
1) tie that pooled buffer to the asynchronous write request
2) keep it locally, so when async request will be completed